Unpin pydantic and numpy in pyproject.toml

lstein opened the pull request:
## What type of PR is this? (check all applicable)

- [ X] Bug Fix


## Have you discussed this change with the InvokeAI team?
- [ X] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [X] Not needed

## Description

Windows users have been getting a lot of OSErrors while installing 3.0.1 during the pip dependency installation phase. Generally the errors have involved just two packages, pydantic and numpy. Looking at the install logs, I see that both of these packages are first installed under one version number by a dependency, and then uninstalled and replaced by a slightly different version specified in invoke's `pyproject.toml`. I think this is the problem - maybe the earlier package is not completely closed before it is uninstalled and reinstalled.

This PR relaxes pinning of numpy and pydantic in `pyproject.toml`. Everything seems to install and run properly. Hopefully it will address the windows install bug as well.

hipsterusername commented:
Have some concerns with letting pydantic get installed via a dependency we don't have pinned. Is it possible we just need to pull pydantic up in the order of install? Copying @Kyle0654 and @psychedelicious for thoughts

lstein commented:
> Have some concerns with letting pydantic get installed via a dependency we don't have pinned. Is it possible we just need to pull pydantic up in the order of install? Copying @Kyle0654 and @psychedelicious for thoughts

Actually pydantic wasn't referenced in `pyproject.toml` in 3.0.0. I added it in 3.0.1, which is what caused the issue. I have now added it back but pinned to `1.*` as suggested by @StAlKeR7779

psychedelicious commented:
If everything works by specifying 1.*, I guess that's fine. 

FWIW, I appear to have had 1.10.10 working just fine since 3.0.0. With pydantic==1.*, I get pydantic 1.10.12.
